server/notification-providers/sip.js

```diff
@@ -0,0 +1,93 @@
+const NotificationProvider = require("./notification-provider");
+
+class SIP extends NotificationProvider {
```

Comment:
Please move the notification provider to a different PR.
Otherwise it is not review

The rest of the code (for example utils) is convoluted enough.